### PR TITLE
Chore: Rewrite cosmos client with osmojs RPC client

### PR DESCRIPTION
**Pull request by akcgjc007**

### CODE REVIEW

File: src/cosmos/CosmosClient.ts [hunk 1]
```diff
@@ -1,68 +1,83 @@
...
+  async getIbcRpcClient() {
+    return await ibc.ClientFactory.createRPCQueryClient({
+      rpcEndpoint: this.rpcEndpoint,
+    });
+  }
+
+  async getCosmwasmRpcClient() {
+    return await cosmwasm.ClientFactory.createRPCQueryClient({
+      rpcEndpoint: this.rpcEndpoint,
+    });
```
Comment (gurbaj5124871) [inline src/cosmos/CosmosClient.ts:81]: Nitpick:
`await` is not needed if the statement is returned directly, also func won't need to have `async` anymore if await is removed from the body

File: src/cosmos/CosmosClient.ts [hunk 2]
```diff
@@ -1,68 +1,83 @@
...
   async validator(validatorAddr: string): Promise<QueryValidatorResponse> {
-    const rpc = await this.getRpcClient();
-    const BankQueryClientImpl = cosmos.staking.v1beta1.QueryClientImpl;
-    const queryServiceBank = new BankQueryClientImpl(rpc);
-    const res = await queryServiceBank.validator({ validatorAddr });
-    return res;
+    const client = await this.getCosmosRpcClient();
+    return await client.cosmos.staking.v1beta1.validator({
+      validatorAddr,
+    });
```
Comment (gurbaj5124871) [inline src/cosmos/CosmosClient.ts:37]:
```suggestion
    return client.cosmos.staking.v1beta1.validator({
      validatorAddr,
    });
```

File: src/cosmos/CosmosClient.ts [hunk 3]
```diff
@@ -1,68 +1,83 @@
...
   async supply(denom: string): Promise<QuerySupplyOfResponse> {
-    const rpc = await this.getRpcClient();
-    const BankQueryClientImpl = cosmos.bank.v1beta1.QueryClientImpl;
-    const queryServiceBank = new BankQueryClientImpl(rpc);
-    const res = await queryServiceBank.supplyOf({ denom });
-    return res;
+    const client = await this.getCosmosRpcClient();
+    return await client.cosmos.bank.v1beta1.supplyOf({
+      denom,
+    });
```
Comment (gurbaj5124871) [inline src/cosmos/CosmosClient.ts:30]:
```suggestion
    const client = await this.getCosmosRpcClient();
    return client.cosmos.bank.v1beta1.supplyOf({
      denom,
    });
```

File: src/cosmos/CosmosClient.ts [hunk 4]
```diff
@@ -1,68 +1,83 @@
...
-    const rpc = await this.getRpcClient();
-    const BankQueryClientImpl = cosmos.bank.v1beta1.QueryClientImpl;
-    const queryServiceBank = new BankQueryClientImpl(rpc);
-    const res = await queryServiceBank.balance({ address, denom });
-    return res;
+    const client = await this.getCosmosRpcClient();
+    return await client.cosmos.bank.v1beta1.balance({
+      address,
+      denom,
+    });
```
Comment (gurbaj5124871) [inline src/cosmos/CosmosClient.ts:23]:
```suggestion
    const client = await this.getCosmosRpcClient();
    return client.cosmos.bank.v1beta1.balance({
      address,
      denom,
    });
```

**Comment by gurbaj5124871** [comment]
Only minor nitpicks review changes, LGTM!

**Review by gurbaj5124871 — DISMISSED** [review]

**Comment by akcgjc007** [comment]
Thanks for the feedback. I have updated the syntax accordingly. 👍 
@gurbaj5124871

**Review by gurbaj5124871 — APPROVED** [review]

**Pull request merged**